.expeditor/scripts/end_to_end/run_e2e_test_core.ps1

```diff
@@ -0,0 +1,126 @@
+    $startTime = Get-Date
+    while (!$TestScript.Invoke()) {
+        $timeTaken = [DateTime]::Now.Subtract($startTime)
+        if ($timeTaken.TotalSeconds -ge $Timeout) {
+            $TimeoutScript.Invoke()
+            break
+        }
+        Start-Sleep -Seconds 1
+    }
+}
+
+function Wait-PathUpdatedAfter($Path, $Time, $Timeout) {
+    $testScript = {
+        if (!(Test-Path -Path $Path)) {
+            return $False
+        }
+        (Get-Item -Path $Path).LastWriteTime -ge $Time
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for '$Path' to be updated after '$($Time.ToString("yyyy-MM-ddTHH:MM:ssZ"))'" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+}
+
+function Test-Port([string]$ComputerName, [int]$Port) {
+    if (Get-Command Test-NetConnection -ErrorAction SilentlyContinue) {
+        (Test-NetConnection -ComputerName $ComputerName -Port $Port).TcpTestSucceeded
+    }
+    else {
+        if ($IsWindows) {
+            Write-Error "`Test-NetConnection` command was not found and `Test-Connection` does not work correctly on Windows"
+        }
+        Test-Connection -ComputerName $ComputerName -TCPPort $Port
+    }
+}
+
+function Wait-Supervisor($Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to start..."
+    $testScript = { Test-Port -ComputerName localhost -Port 9631 }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for Supervisor to start" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+    Write-Host "Supervisor is now running."
+}
+
+function Start-Supervisor($Timeout = 1) {
+    hab sup run --no-color &
+    Wait-Supervisor -Timeout $Timeout
+}
+
+function Wait-SupervisorService($ServiceName, $Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to start $ServiceName ..."
+    $testScript = { 
+        try {
+            $status = (Invoke-WebRequest "http://localhost:9631/services/$ServiceName/default" |
+                ConvertFrom-Json).process.state
+            $status -eq "up"
+        }
+        catch { } # We ignore 404s and other unsuccesful codes
+    }
+    $timeoutScript = { Write-Error "Timed out waiting $Timeout seconds for Supervisor to start $ServiceName" }
+    Wait-True -TestScript $testScript -TimeoutScript $timeoutScript -Timeout $Timeout
+    Write-Host "$ServiceName is now up."
+}
+  
+function Install-SupervisorService($PackageName, $Timeout = 1) {
+    $svcName = ($PackageName -split "/")[1]
+    $_ = hab svc load $PackageName
+    Wait-SupervisorService $svcName -Timeout $Timeout
+    $svcName
+}
+  
+function Wait-SupervisorServiceUnload($ServiceName, $Timeout = 1) {
+    Write-Host "Waiting up to $Timeout seconds for Supervisor to unload $ServiceName ..."
+    $testScript = { 
+        try {
+            Invoke-WebRequest "http://localhost:9631/services/$ServiceName/default"
```

Comment:
I think you will want to add a `$false` after the `Invoke-WebRequest` because if it succeeds it will return the response.